api/v1alpha1/stage_types.go

```diff
@@ -128,6 +128,14 @@ type FreightOriginKind string
 
 const FreightOriginKindWarehouse FreightOriginKind = "Warehouse"
 
+// +kubebuilder:validation:Enum={AllUpstream,AnyUpstream}
+type FreightAvailabilityStrategy string
+
+const (
+	FreightAvailabilityStrategyAllUpstream FreightAvailabilityStrategy = "AllUpstream"
+	FreightAvailabilityStrategyAnyUpstream FreightAvailabilityStrategy = "AnyUpstream"
+)
```

Comment:
Wonder if brevity might be an improvement here, but open to discussion.

```suggestion
const (
	FreightAvailabilityStrategyAll FreightAvailabilityStrategy = "All"
	FreightAvailabilityStrategyOneOf FreightAvailabilityStrategy = "OneOf"
)
```

Reply:
The explicit use of 'Upstream' in these enums was chosen in case there was a future availability strategy that wasn't referring to upstream Stage state, but practically those use cases will probably need to be implemented as specific values like `RequiredSoakTime` as it will have some kind of configurable 'value' (ie, the soak time).

Maybe you have more context to guide whether that can be a future concern (if at all).  I've taken your suggestion though and applied it to the PR 👍

Reply:
I have a hard time imagining what enums we would introduce that would not directly relate to stages, at this point. The only possible thing I could think of is to move the `direct` field to an enum selector, but I fear that ship has already sailed.

This makes me wonder however if `FreightAvailabilityStrategy` is the right name for the field/type, or if it should hint towards something like `StageSelectionStrategy`. But I think @krancour is a better judge here, as English isn't my primary language.